OneCommunityGlobal/HighestGoodNetworkApp
# Lucas user hours set to zero

This PR is due to the following situation:
(PRIORITY LOW) Jae: Special Situation - User Hours Set to Zero - (WIP: Lucas)
In some cases a user’s hours will be set to Zero. This is usually when a user has gotten a job and is just wrapping up their work with us. In those cases, I’d like them not to contribute to Dashboard → Leaderboard → HGN Totals number (number of users). This is needed so that the number of “Users” in the app properly reflects those who are adhering to and meeting the One Community minimums for being a team member. 
The same should be true for User Class “Mentor” when it is added
b.    I still need these people’s emails listed at the bottom of the weekly reports though, in case I need to use that list to email the entire team.

<hr/>

### Main changes

Mentors and users with zero hours now do not appear on the Leaderboard. Only Onwers, Administrators and Core Team users can see them on the Leaderboard. Mentors and users with zero hours can see themselves and team members on the Leaderboard. Mentor users and zero hour users no longer add to the HGN total members.

<hr/>

### How to test
Use the backend [#293](https://github.com/OneCommunityGlobal/HGNRest/pull/293) PR to test this PR.

- checkout the current branch, both in backend and frontend
- npm install (on the backend)
- npm run build (on the backend)
- npm run start (on the backend)
- npm install (on the frontend)
- npm run start:local (on the frontend)
- Login with various roles and teams and see if the user can see the mentor users or zero hour users (only owners, administrators and core teams can see them. Mentor users and zero hours users can see themselves and team members)
- Check if changing a user's hours to zero or the role to Mentor change the HGN total members number
